firestore/angular-rewrite/functions/src/index.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    onDocumentWritten,
+} from "firebase-functions/v2/firestore";
+import * as logger from "firebase-functions/logger";
+import { getFirestore } from "firebase-admin/firestore";
+import { initializeApp } from "firebase-admin/app";
+
+import { Restaurant } from '../../types/restaurant';
+import { Rating } from '../../types/ratings';
+import { strict as assert } from 'assert';
+
+initializeApp();
+const db = getFirestore();
+
+export const updateNumRatings = onDocumentWritten(
+    "restaurants/{restaurtantID}/ratings/{ratingID}", async (event) => {
+        logger.info(
+            `Updating the numRatings and avgRatings for restuarant ` +
+            `${event.params.restaurtantID}`);
+
+        // Get num reviews from restaurant and compare to actual num reviews
+        const restuarantDocRef = db.doc(
+            `restaurants/${event.params.restaurtantID}`);
+
+        logger.info(`Fetching data for restaurant ` +
+            `${event.params.restaurtantID}`);
+        const restaurantDocFromFirebase = await restuarantDocRef.get();
+        const restaurantData = restaurantDocFromFirebase.data() as Restaurant;
+        const numRatingsReported = restaurantData.numRatings;
+        const fetchedRatingDocs = (await db.collection(`restaurants/${event.params.restaurtantID}/ratings`).get()).docs
+        const actualRatings: Rating[] = []
+        fetchedRatingDocs.forEach(rating => actualRatings.push(rating.data() as Rating))
```

Comment:
Let's use a .map() instead of forEach and we can assign directly to a variable. Something like this:

`const actualRatings: Rating[] = fetchedRatingDocs.map(ratingDoc => return ratingDoc.data() as Rating);`